contrib/epee/include/misc_language.h

```diff
@@ -30,74 +30,14 @@
 
 #include <boost/utility/value_init.hpp>
 #include <boost/shared_ptr.hpp>
-#include <limits>
-#include <functional>
 #include <vector>
 namespace epee
 {
-#define STD_TRY_BEGIN() try {
-
-#define STD_TRY_CATCH(where_, ret_val) \
-	} \
-	catch (const std::exception  &e) \
-	{ \
-		LOG_ERROR("EXCEPTION: " << where_  << ", mes: "<< e.what());  \
-		return ret_val; \
-	} \
-	catch (...) \
-	{ \
-		LOG_ERROR("EXCEPTION: " << where_ ); \
-		return ret_val; \
-	}
-
-
-
 #define AUTO_VAL_INIT(v)   boost::value_initialized<decltype(v)>()
 
 namespace misc_utils
 {
-	template<typename t_type>
-		t_type get_max_t_val(t_type t)
```

Comment:
I'd leave it.

Reply:
Why use `get_max_t_val` instead of `std::numeric_limits<t_type>::max()` directly? `std::numeric_limits` is already used 148 times throughout the project and IMO its confusing to pass in a raw value and receive a static limit unrelated to the value passed to the function.

Reply:
As far as I see it evaluates to the same type and value.

It's not a state of the art function though anyway.

Reply:
Yeah it should evaluate to to the same value, but it feels clunkier to use IMO because you pass a raw value of the type that you want to get the limit of, instead of just asking for it based on type.